apis/v1alpha1/httproute_types.go

```diff
@@ -58,6 +58,26 @@ type HTTPRouteHost struct {
 	// +optional
 	Hostnames []string `json:"hostnames,omitempty"`
 
+	// TLS defines the TLS certificate to use for Hostnames defined in this
+	// Route. This configuration only takes effect if the AllowRouteOverride
+	// field is set to true in the associated Gateway resource.
+	//
+	// Collisions can happen if multiple HTTPRoutes define a TLS certificate
+	// for the same hostname. The behavior in such a case is implementation
```

Comment:
Does this behavior need to be implementation specific? Could we rely on some combination of most specific hostname match + oldest resource wins? If there's a conflict within a route maybe we could then say that the first matching rule in that route wins?

Reply:
> Could we rely on some combination of most specific hostname match + oldest resource wins?

This comment applies when the hostnames are equal so most specific hostname doesn't matter here. The oldest resource wins makes sense here.

We have had users who have an edge case where the oldest resource has an expired certificate and they expected the controller to fallback to the next valid certificate in that case. While this is technically possible, it is more complicated than I'd like the controller to be. Is that fair?

> If there's a conflict within a route maybe we could then say that the first matching rule in that route wins?

SGTM.